src/Http/Client/ClientEvent.php

```diff
@@ -0,0 +1,120 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         5.1.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Http\Client;
+
+use Cake\Event\Event;
+use Cake\Http\Client;
+use InvalidArgumentException;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Class Client Event
+ *
+ * @extends \Cake\Event\Event<\Cake\Http\Client>
+ */
+class ClientEvent extends Event
```

Comment:
This is our first foray into custom events. If this goes well we could consider it in other parts of the framework. I could see it helping provide better type/runtime safety with event payloads.